operator/pkg/nodeipam/nodesvclb.go

```diff
@@ -172,6 +186,43 @@ func (r *nodeSvcLBReconciler) getRelevantNodes(ctx context.Context, svc *corev1.
 		return nil, err
 	}
 
+	var nodes corev1.NodeList
+
+	// Special handling for "virtual" Endpoints like Gateways and Ingresses
+	// If the Service has either label that Cilium injects, use all nodes, or filter by Annotation value
+	// https://github.com/cilium/cilium/pull/30038#issuecomment-1970423094
```

Comment:
As I was saying in the other thread I am not in favor of a special treatment for Cilium ingress/GatewayAPI, a general annotation enabling this behavior of treating all the nodes (+ also checking that `externalTrafficPolicy` is set on `Cluster` in that case) would be better IMO.

Reply:
Actually if the behavior would be to have an annotation enabling this behavior + considering this enabled as well for Cilium Ingress/GatewayAPI it could be nice as well to lighten up a bit the "config" that users have to do.

Reply:
So if I understand correctly, you'd be in favor of refactoring the existing code to have a single handler which would listen on all nodes by default and then limit nodes based on the new annotation?

Reply:
What I would envision is:
- `io.cilium.nodeipam/match-node-labels`: the same as your implementation except that it should filters on all code path
- `io.cilium.nodeipam/clusterwide` (or another better name, if someone has a better idea): If enabled it does not looks at the backing pods/endpointslice and treat all the nodes. Disabled by default but could be "auto enabled" if detecting that it's a ingress/gateway api Cilium svc. Also errors out if enabled and `externalClusterTrafficPolicy` is set to `Local`.

WDYT?